### PR TITLE
Feature/faqs order

### PR DESCRIPTION
**Pull request by binarymelon**
Initial pass at faqs index.  This just allows manually setting an index for the faq.  Next step will be adding up/down arrows on Admin index to reorder.

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Looks pretty good. I think the only thing I'd offer is maybe change the name from "index" to something like "order" or "position", as the word "index" is very tied to SQL indexes and might get confusing.

File: app/Faq.php [hunk 1]
```diff
@@ -9,10 +9,11 @@
  * @package App
  * @property string $question
  * @property text $answer
+ * @property integer $index
```
Comment (x3N1GM4x) [inline app/Faq.php]:
```suggestion
 * @property integer $position
```

File: app/Faq.php [hunk 2]
```diff
@@ -9,10 +9,11 @@
...
 */
 class Faq extends Model
 {
-    protected $fillable = ['question', 'answer'];
+    protected $fillable = ['question', 'answer', 'index'];
```
Comment (x3N1GM4x) [inline app/Faq.php]:
```suggestion
    protected $fillable = ['question', 'answer', 'position'];
```

File: app/Http/Controllers/Admin/FAQsController.php [hunk 3]
```diff
@@ -28,7 +28,7 @@ public function index()
             }
             $faqs = Faq::onlyTrashed()->get();
         } else {
-            $faqs = Faq::all();
+            $faqs = Faq::orderBy('index', 'asc')->get();
```
Comment (x3N1GM4x) [inline app/Http/Controllers/Admin/FAQsController.php]:
```suggestion
            $faqs = Faq::orderBy('position', 'asc')->get();
```

File: app/Http/ViewComposers/FAQComposer.php [hunk 4]
```diff
@@ -28,6 +28,6 @@ public function __construct(Faq $faqs)
      */
     public function compose(View $view)
     {
-        $view->faqs = $this->faqs->all();
+        $view->faqs = $this->faqs->orderBy('index', 'asc')->get();
```
Comment (x3N1GM4x) [inline app/Http/ViewComposers/FAQComposer.php]:
```suggestion
        $view->faqs = $this->faqs->orderBy('position', 'asc')->get();
```

File: database/migrations/2018_12_10_234558_add_index_to_faqs_table.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+      Schema::table('faqs', function (Blueprint $table) {
+          $table->integer('index')->default(9999);
```
Comment (x3N1GM4x) [inline database/migrations/2018_12_10_234558_add_index_to_faqs_table.php]:
```suggestion
          $table->integer('position')->default(9999);
```

File: database/migrations/2018_12_10_234558_add_index_to_faqs_table.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      });
+
+      \DB::statement('UPDATE faqs SET `index` = id');
```
Comment (x3N1GM4x) [inline database/migrations/2018_12_10_234558_add_index_to_faqs_table.php]:
```suggestion
      \DB::statement('UPDATE faqs SET `position` = id');
```

File: database/migrations/2018_12_10_234558_add_index_to_faqs_table.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+      Schema::table('faqs', function (Blueprint $table) {
+          $table->dropColumn('index');
```
Comment (x3N1GM4x) [inline database/migrations/2018_12_10_234558_add_index_to_faqs_table.php]:
```suggestion
          $table->dropColumn('position');
```

File: resources/views/admin/faqs/create.blade.php [hunk 8]
```diff
@@ -34,6 +34,18 @@
                     @endif
                 </div>
             </div>
+            <div class="row">
+                <div class="col-xs-12 form-group">
+                    {!! Form::label('index', trans('quickadmin.faqs.fields.index').'*', ['class' => 'control-label']) !!}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/create.blade.php]:
```suggestion
                    {!! Form::label('position', trans('quickadmin.faqs.fields.position').'*', ['class' => 'control-label']) !!}
```

File: resources/views/admin/faqs/create.blade.php [hunk 9]
```diff
@@ -34,6 +34,18 @@
...
+                    {!! Form::number('index', old('index'), ['class' => 'form-control', 'placeholder' => '', 'required' => '']) !!}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/create.blade.php]:
```suggestion
                    {!! Form::number('position', old('position'), ['class' => 'form-control', 'placeholder' => '', 'required' => '']) !!}
```

File: resources/views/admin/faqs/create.blade.php [hunk 10]
```diff
@@ -34,6 +34,18 @@
...
+                    <p class="help-block"></p>
+                    @if($errors->has('index'))
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/create.blade.php]:
```suggestion
                    @if($errors->has('position'))
```

File: resources/views/admin/faqs/create.blade.php [hunk 11]
```diff
@@ -34,6 +34,18 @@
...
+                        <p class="help-block">
+                            {{ $errors->first('index') }}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/create.blade.php]:
```suggestion
                            {{ $errors->first('position') }}
```

File: resources/views/admin/faqs/edit.blade.php [hunk 12]
```diff
@@ -35,6 +35,18 @@
                     @endif
                 </div>
             </div>
+            <div class="row">
+                <div class="col-xs-12 form-group">
+                    {!! Form::label('index', trans('quickadmin.faqs.fields.index').'*', ['class' => 'control-label']) !!}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/edit.blade.php]:
```suggestion
                    {!! Form::label('position', trans('quickadmin.faqs.fields.position').'*', ['class' => 'control-label']) !!}
```

File: resources/views/admin/faqs/edit.blade.php [hunk 13]
```diff
@@ -35,6 +35,18 @@
...
+                    {!! Form::number('index', old('index'), ['class' => 'form-control', 'placeholder' => '', 'required' => '']) !!}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/edit.blade.php]:
```suggestion
                    {!! Form::number('position', old('position'), ['class' => 'form-control', 'placeholder' => '', 'required' => '']) !!}
```

File: resources/views/admin/faqs/edit.blade.php [hunk 14]
```diff
@@ -35,6 +35,18 @@
...
+                    <p class="help-block"></p>
+                    @if($errors->has('index'))
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/edit.blade.php]:
```suggestion
                    @if($errors->has('position'))
```

File: resources/views/admin/faqs/edit.blade.php [hunk 15]
```diff
@@ -35,6 +35,18 @@
...
+                        <p class="help-block">
+                            {{ $errors->first('index') }}
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/edit.blade.php]:
```suggestion
                            {{ $errors->first('position') }}
```

File: resources/views/admin/faqs/index.blade.php [hunk 16]
```diff
@@ -26,11 +26,12 @@
                         @endcan
 
                         <th>@lang('quickadmin.faqs.fields.question')</th>
+                        <th>@lang('quickadmin.faqs.fields.index')</th>
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/index.blade.php]:
```suggestion
                        <th>@lang('quickadmin.faqs.fields.position')</th>
```

File: resources/views/admin/faqs/index.blade.php [hunk 17]
```diff
@@ -40,6 +41,7 @@
                                 @endcan
 
                                 <td field-key='title'>{{ $faq->question }}</td>
+                                <td field-key='index'>{{ $faq->index }}</td>
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/index.blade.php]:
```suggestion
                                <td field-key='position'>{{ $faq->position}}</td>
```

File: resources/views/admin/faqs/show.blade.php [hunk 18]
```diff
@@ -20,6 +20,10 @@
                             <th>@lang('quickadmin.faqs.fields.answer')</th>
                             <td field-key='conteanswernt'>{!! $faq->answer !!}</td>
                         </tr>
+                        <tr>
+                            <th>@lang('quickadmin.faqs.fields.index')</th>
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/show.blade.php]:
```suggestion
                            <th>@lang('quickadmin.faqs.fields.position')</th>
```

File: resources/views/admin/faqs/show.blade.php [hunk 19]
```diff
@@ -20,6 +20,10 @@
...
+                            <td field-key='conteanswernt'>{!! $faq->index !!}</td>
```
Comment (x3N1GM4x) [inline resources/views/admin/faqs/show.blade.php]:
```suggestion
                            <td field-key='conteanswernt'>{!! $faq->position!!}</td>
```

**Comment by x3N1GM4x** [comment]
Tested locally and all seems good! Thanks @binarymelon.

**Comment by x3N1GM4x** [comment]
Deployed to production.

**Comment by x3N1GM4x** [comment]
[P2: Implement ability to order FAQs](https://trello.com/c/CR0eXrTA/58-p2-implement-ability-to-order-faqs)

**Pull request merged**